openedx/core/djangoapps/content/learning_sequences/migrations/0001_initial.py

```diff
@@ -0,0 +1,125 @@
+                ('modified', model_utils.fields.AutoLastModifiedField(default=django.utils.timezone.now, editable=False, verbose_name='modified')),
+            ],
+        ),
+        migrations.AddIndex(
+            model_name='learningcontext',
+            index=models.Index(fields=['-published_at'], name='learning_se_publish_62319b_idx'),
+        ),
+        migrations.AddField(
+            model_name='coursesectionsequence',
+            name='learning_context',
+            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='section_sequences', to='learning_sequences.LearningContext'),
+        ),
+        migrations.AddField(
+            model_name='coursesectionsequence',
+            name='section',
+            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='learning_sequences.CourseSection'),
+        ),
+        migrations.AddField(
+            model_name='coursesectionsequence',
+            name='sequence',
+            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='learning_sequences.LearningSequence'),
+        ),
+        migrations.AddField(
+            model_name='coursesection',
+            name='learning_context',
+            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='sections', to='learning_sequences.LearningContext'),
+        ),
+        migrations.AlterUniqueTogether(
+            name='learningsequence',
+            unique_together={('learning_context', 'usage_key')},
+        ),
+        migrations.AlterUniqueTogether(
+            name='coursesectionsequence',
+            unique_together={('learning_context', 'order')},
+        ),
+        migrations.AlterUniqueTogether(
+            name='coursesection',
+            unique_together={('learning_context', 'usage_key')},
+        ),
+        migrations.AlterIndexTogether(
+            name='coursesection',
+            index_together={('learning_context', 'order')},
+        ),
+
+        # Custom code: Convert tables to utf8_bin because we want to allow case-
+        # sensitive comparisons for things like UsageKeys, CourseKeys, and
+        # slugs.
+        #
+        migrations.RunSQL(
+            'ALTER TABLE learning_sequences_learningcontext MODIFY context_key VARCHAR(255) CHARACTER SET utf8 COLLATE utf8_bin;',
+            reverse_sql=migrations.RunSQL.noop,
+        ),
+        migrations.RunSQL(
+            'ALTER TABLE learning_sequences_coursesection MODIFY usage_key VARCHAR(255) CHARACTER SET utf8 COLLATE utf8_bin;',
+            reverse_sql=migrations.RunSQL.noop,
+        ),
+        migrations.RunSQL(
+            'ALTER TABLE learning_sequences_learningsequence MODIFY usage_key VARCHAR(255) CHARACTER SET utf8 COLLATE utf8_bin;',
+            reverse_sql=migrations.RunSQL.noop,
+        ),
+    ]
```

Comment:
@macdiesel, @doctoryes: In order to avoid some of the headaches that we were involved with today and last week w.r.t. case-insensitivity on CourseKey columns, I'm manually munged the migration for these new models to convert their collation to utf8_bin so it's case sensitive by default. Please let me know if you have any concerns.

FYI @cpennington

Reply:
Decided in data-eng office hours that this was okay to do.